docs/en/operations/settings/settings.md

```diff
@@ -509,6 +509,23 @@ Possible values:
 
 Default value: `ALL`.
 
+## join_algorithm {#settings-join_algorithm}
+
+Specifies [JOIN](../../sql-reference/statements/select/join.md) algorithm.
+
+Possible values:
+
+- 'hash' — [Hash join algorithm](https://en.wikipedia.org/wiki/Hash_join) is used.
+- 'partial_merge' — [Sort-merge algorithm](https://en.wikipedia.org/wiki/Sort-merge_join) is used.
+- 'prefer_partial_merge' — ClickHouse always tries to use `merge` join if possible.
+- 'auto' — ClickHouse tries to change `hash` join to `merge` join on the fly to avoid out of memory.
```

Comment:
```suggestion
- `hash` — [Hash join algorithm](https://en.wikipedia.org/wiki/Hash_join) is used.
- `partial_merge` — [Sort-merge algorithm](https://en.wikipedia.org/wiki/Sort-merge_join) is used.
- `prefer_partial_merge` — ClickHouse always tries to use `merge` join if possible.
- `auto` — ClickHouse tries to change `hash` join to `merge` join on the fly to avoid out of memory.
```